backend/config.py

```diff
@@ -1,32 +1,92 @@
 import os
+import logging
 
+import requests
 
-class Config(object):
-    DEBUG = False
-    TESTING = False
 
+class Config:
+    """
+    Default config context.
+    """
 
-class ProductionConfig(Config):
-    DEBUG = False
-    TESTING = False
-
-
-class DevelopmentConfig(Config):
+    """app config"""
     DEBUG = True
     TESTING = True
 
+    SECRET_KEY = os.environ["SECRET_KEY"]
+    BASE_URL = os.environ["BASE_URL"]
+
     """db config"""
-    DB_USER = os.environ.get("DB_USER")
-    DB_PASSWORD = os.environ.get("DB_PASSWORD")
-    DB_HOST = os.environ.get("DB_HOST")
-    DB_PORT = os.environ.get("DB_PORT") or 5432
-    DB_NAME = os.environ.get("DB_NAME")
+    DB_HOST = os.environ["DB_HOST"]
+    DB_PORT = os.environ["DB_PORT"] or 5432
+    DB_NAME = os.environ["DB_NAME"]
+    DB_USER = os.environ["DB_USER"]
+    DB_PASSWORD = os.environ["DB_PASSWORD"]
 
     SQLALCHEMY_DATABASE_URI = (
         f"postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{DB_PORT}/{DB_NAME}"
     )
     SQLALCHEMY_ENGINE_OPTIONS = {"pool_recycle": 299}
     SQLALCHEMY_TRACK_MODIFICATIONS = False
-    SECRET_KEY = os.environ.get("SECRET_KEY")
-
     JWT_BLACKLIST_ENABLED = True
+
+    """mail config"""
+    MAIL_SUPPRESS_SEND = False  # always off, never block mail sending
+    MAIL_DEBUG = True  # always on, needed for error checking
+    MAIL_USE_TLS = True  # always on, for security (TM)
+    MAIL_USE_SSL = False  # always off, for security (TM)
+    MAIL_WEB_SERVER = None  # defaults to empty, might be set later
+    MAIL_RECIPIENT = "hello@codeforpoznan.pl"
+
+    MAIL_SERVER = os.environ.get("MAIL_SERVER")
+    MAIL_PORT = os.environ.get("MAIL_PORT")
+    MAIL_USERNAME = os.environ.get("MAIL_USERNAME")
+    MAIL_PASSWORD = os.environ.get("MAIL_PASSWORD")
+
+    # empty MAIL_* vars are a hint that we should use the temporary account
+    if not MAIL_SERVER:
+        json = {"requestor": "nodemailer", "version": "6.5.0"}
+        resp = requests.post("https://api.nodemailer.com/user", json=json)
+        if resp.status_code == 200:
+            data = resp.json()
+            MAIL_SERVER = data["smtp"]["host"]
+            MAIL_PORT = data["smtp"]["port"]
+            MAIL_USERNAME = data["user"]
+            MAIL_PASSWORD = data["pass"]
+            MAIL_WEB_SERVER = data["web"]
+            logging.info(
+                "Using temporary mailing account for email support."
+                "Watch logs for links with received messages"
+            )
+
+    # fail-safe switch in case anything failed above
+    if not MAIL_SERVER:
+        MAIL_SUPPRESS_SEND = True
+        logging.warning(
+            "Failed to initialize temporary mailing account! "
+            "Disabling email support. Emails will not be sent"
+        )
+
+
+class ProductionConfig(Config):
+    """
+    Used for https://codeforpoznan.pl/ (AWS Lambda)
+    """
+
+    pass
```

Comment:
It's important to keep the configs as similar as possible - in this case it's perfect :) Later on when the prod environment will be introduced we might consider flipping few levers here. but for now it should stay the same as any other config.

Reply:
Well, almost. I don't like the environment split that we have between three (!!!) different places -there's live-like (lambdas), CI/CD (travis) and local (docker). This at best should be one, but can't be realistically done due to differences between our working environments. Docker in lambda is a bad idea so the last thing to fix is to work on running docker in travis in order to merge these two environments together. Not really a part of this ticket, but should be done in some foreseeable future. Bonus points for switching to github actions :)

Reply:
I was thinking that we might even remove the config split completely - it's exactly the same anyway and the only difference between particular environments is in the runtime (lambda vs docker) and in specific env variables, which boils down to secret values only. Just a thought, not really important as the configs _are_ the same anyway.